include/StochasticNoiseSimulator.hpp

```diff
@@ -3,189 +3,142 @@
+        Simulator<DDPackage>::dd->resize(qc->getNqubits());
+        setMaxInstances();
     }
 
     StochasticNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc,
-                             const std::string&                       noise_effects,
-                             double                                   noise_prob,
-                             long                                     stoch_runs,
-                             unsigned int                             step_number,
-                             double                                   step_fidelity,
-                             const std::string&                       recorded_properties):
-        qc(qc),
-        step_number(step_number), step_fidelity(step_fidelity) {
-        setNoiseEffects(noise_effects);
-        setRecordedProperties(recorded_properties);
-        initializeNoiseProbabilities(noise_prob);
-        setNumberOfRuns(stoch_runs);
-    }
+                             const std::string&                       cNoiseEffects,
+                             double                                   cNoiseProbability,
+                             std::optional<double>                    cAmpDampingProbability,
+                             double                                   cMultiQubitGateFactor,
+                             std::size_t                              cStochRuns,
+                             const std::string&                       recordedProperties,
+                             bool                                     unoptimizedSim,
+                             unsigned int                             stepNumber,
+                             double                                   stepFidelity,
+                             std::size_t                              seed = 0U):
+        Simulator<DDPackage>(seed),
+        qc(qc), stepNumber(stepNumber), stepFidelity(stepFidelity) {
+        // setNoiseEffects
+        for (const auto noise: cNoiseEffects) {
+            switch (noise) {
+                case 'A':
+                    noiseEffects.push_back(dd::amplitudeDamping);
+                    break;
+                case 'P':
+                    noiseEffects.push_back(dd::phaseFlip);
+                    break;
+                case 'D':
+                    noiseEffects.push_back(dd::depolarization);
+                    break;
+                default:
+                    throw std::runtime_error("Unknown noise operation '" + cNoiseEffects + "'\n");
+            }
+        }
 
-    std::map<std::string, std::size_t> Simulate(unsigned int shots) override;
+        // initializeNoiseProbabilities
+        noiseProbability = cNoiseProbability;
+        // Default value for amplitude damping prob is double the general error probability
+        amplitudeDampingProb = (cAmpDampingProbability) ? cAmpDampingProbability.value() : noiseProbability * 2;
+        multiQubitGateFactor = cMultiQubitGateFactor;
```

Comment:
already set these in the initialiser lists of the constructor just as in the deterministic simulator